install/install.ps1

```diff
@@ -1,94 +1,203 @@
+    $zipFile = ([System.IO.Path]::Combine($azureauthDirectory, $releaseFile))
+
+    Write-Verbose "Creating ${azureauthDirectory}"
+    $null = New-Item -ItemType Directory -Force -Path $azureauthDirectory
+
+
+    Write-Verbose "Downloading ${releaseUrl} to ${zipFile}"
+    $client = New-Object System.Net.WebClient
+    $client.DownloadFile($releaseUrl, $zipFile)
+
+    # A running instance of azureauth can cause installation to fail, so we try to kill any running instances first.
+    # We suppress taskkill output here because this is a best effort attempt and we don't want the user to see its output.
+    # Here, Get-Process is used to first determine whether there is an existing azureauth process. If there is, kill the existing process first.
+    $ProcessCheck = Get-Process -Name azureauth -ErrorAction SilentlyContinue -ErrorVariable ProcessError
+    if ($null -ne $ProcessCheck) {
+        Write-Verbose "Stopping any currently running azureauth instances"
+        taskkill /f /im azureauth.exe 2>&1 | Out-Null
+
+        # After killing the process it is still possible for there there to be locks on the files it was using (including
+        # its own DLLs). The OS may take an indeterminate amount of time to clean those up, but so far we've observed 1
+        # second to be enough.
+        Start-Sleep -Seconds 1
+    }
+
+    if (Test-Path -Path $targetDirectory) {
+        Write-Verbose "Removing pre-existing extracted directory at ${targetDirectory}"
+        Remove-Item -Force -Recurse $targetDirectory
+    }
+
+    Write-Verbose "Extracting ${zipFile} to ${targetDirectory}"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $targetDirectory)
+
+    Write-Verbose "Removing ${zipFile}"
+    Remove-Item -Force $zipFile
+
+    if ($NoUpdatePath) {
+        Write-Verbose "Not updating `$env:PATH"
+    }
+    else{
+        $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+        $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+        $currentAzureauth = (get-command azureauth -ErrorAction SilentlyContinue).Source
+        $currentAzureauthParent = if($null -ne $currentAzureauth) {
+            (get-item $currentAzureauth).Directory.Parent.FullName }
+        
+        $newPath = "";
+
+        if (($null -ne $currentPath) -And ($currentPath.Contains($azureauthDirectory) -Or $currentPath.Contains($currentAzureauthParent))) {
+            $paths = $currentPath.Split(";")
+            $pathArr = @()
+            ForEach($path in $paths){
+                if(!(($path.Equals("")) -Or ($path.Contains($azureauthDirectory)) -Or (($null -ne $currentAzureauthParent) -And $currentAzureauthParent.Contains($path)))){
+                    $pathArr += "${path}"
+                }
+                else {
+                    Write-Verbose "Removing '${path}' from `$env:PATH"
+                }

```

Comment:
This code block has less readability than others. Think about make it simple or extract a pattern to an extra method.

Reply:
I am interested in a recommendation to make it easier to read. Is it mainly the consecutive `-Or` patterns that harms readability for you, @goagain? Extracting this to a separate function is possible, but tricky because of the number of local variables referenced.